### PR TITLE
[modify .idea/misc.xml and AuthController.java]

### PR DESCRIPTION
**Pull request by samuelprasad-yubi**
### Title: IDE Configuration Update for AdvancedJavaMasterClass Project

### Description:
This pull request introduces updates to the `.idea/misc.xml` file and the `AuthController.java` class in the `easy-auth-application` package, respectively. The changes aim to modify some configurations and fix a typo error in the codebase.

### Changes Made:
1. In the `.idea/misc.xml`, updated the project's default language level from JDK_21 to an unspecified version (JDK_X). This change is likely for future compatibility with newer Java SDK versions.
2. In the `AuthController.java` file, corrected a typo error in setting the user role by changing "ROLE\_ADMIN" and "ROLE\_USER" to "ROLE\_ADMIN" and "ROLE\_USE", respectively. This ensures consistent naming for roles throughout the application.
3. In `AuthController.java`, updated the JavaDoc comment style for better readability and consistency.

#### Code Quality:
The changes made in this PR do not significantly impact code quality but rather focus on configuration updates and a typo correction. However, it is essential to ensure that all other components of the project adhere to the coding standards before merging this PR.

#### Functionality:
No new functionality was added or changed with these modifications. The primary goal was to correct an error in the codebase and update some configurations for potential future compatibility.

### Code Review Comments:
1. In `AuthController.java`, line 20, consider adding a Javadoc comment explaining the purpose of this block of code.
   ```java
   if (signUpRequest.getRole() == null || signUpRequest.getRole().equalsIgnoreCase("user")) {
       user.setRole("ROLE_USE"); // Set role to "USER" instead of "ROLE\_USE". Consider renaming the constant as well.
       ...
   } else if (signUpRequest.getRole().equalsIgnoreCase("admin")) {
       user.setRole("ROLE_ADMIN");
       ...
   } else {
       // Handle other role cases here.
   }
   ```
2. In `AuthController.java`, line 13, consider adding a Javadoc comment explaining the purpose of this method and its expected input/output behavior.
   ```java
   @PostMapping("/signup")
   public ResponseEntity<?> signUp(@RequestBody SignUpRequest signUpRequest) {
       // Method implementation here.
   }
   ```
3. In `AuthController.java`, consider adding a constant for the user role and renaming all occurrences of "ROLE\_ADMIN" and "ROLE\_USER" to these constants throughout the codebase, ensuring consistency in naming conventions.
4. Ensure that there are no other instances where the incorrect role names have been used by checking other related classes or methods within the project.
5. In `AuthController.java`, consider adding input validation checks for the `SignUpRequest` object to ensure data integrity and prevent potential security vulnerabilities.
6. Review all other parts of the codebase, including tests and documentation, to ensure they are up-to-date with these changes.
7. Consider running a static analysis tool like Checkstyle or SonarQube on the project to identify any coding issues that may have been missed during development.
8. Ensure that there is proper error handling in place for cases where input validation fails, and users receive appropriate error messages instead of generic errors.
9. Review other parts of the application's codebase, such as controllers, services, or models, to ensure consistency with these changes.
10. Consider adding unit tests for this method to cover edge cases and validate its behavior thoroughly.

### CODE REVIEW

File: easy-auth-application/src/main/java/com/jmca/easyauthapplication/controller/AuthController.java [hunk 1]
```diff
@@ -78,7 +78,7 @@ public ResponseEntity<?> registerUser(@Valid @RequestBody SignupRequest signUpRe
                         encoder.encode(signUpRequest.getPassword()));
 
         if (signUpRequest.getRole() == null || signUpRequest.getRole().equalsIgnoreCase("user")) {
-            user.setRole("ROLE_USER");
+            user.setRole("ROLE_USE");
```
Comment (samuelprasad-yubi) [inline easy-auth-application/src/main/java/com/jmca/easyauthapplication/controller/AuthController.java:81]: @reviewzen suggestion

Comment (samuelprasad-yubi) [inline easy-auth-application/src/main/java/com/jmca/easyauthapplication/controller/AuthController.java:81]: @reviewzen suggestion

Comment (samuelprasad-yubi) [inline easy-auth-application/src/main/java/com/jmca/easyauthapplication/controller/AuthController.java:81]: @reviewzen suggestion

Comment (samuelprasad-yubi) [inline easy-auth-application/src/main/java/com/jmca/easyauthapplication/controller/AuthController.java]: @reviewzen suggestion

Comment (samuelprasad-yubi) [reply easy-auth-application/src/main/java/com/jmca/easyauthapplication/controller/AuthController.java]: @reviewzen suggestion

File: easy-auth-application/src/main/java/com/jmca/easyauthapplication/controller/AuthController.java [hunk 2]
```diff
@@ -78,7 +78,7 @@ public ResponseEntity<?> registerUser(@Valid @RequestBody SignupRequest signUpRe
...
         } else if (signUpRequest.getRole().equalsIgnoreCase("admin")) {
             user.setRole("ROLE_ADMIN");
         } else {
```
Comment (samuelprasad-yubi) [inline easy-auth-application/src/main/java/com/jmca/easyauthapplication/controller/AuthController.java:84]: @reviewzen suggestion

File: easy-auth-application/src/main/java/com/jmca/easyauthapplication/controller/AuthController.java [hunk 3]
```diff
@@ -78,7 +78,7 @@ public ResponseEntity<?> registerUser(@Valid @RequestBody SignupRequest signUpRe
                         encoder.encode(signUpRequest.getPassword()));
 
         if (signUpRequest.getRole() == null || signUpRequest.getRole().equalsIgnoreCase("user")) {
-            user.setRole("ROLE_USER");
+            user.setRole("ROLE_USE");
         } else if (signUpRequest.getRole().equalsIgnoreCase("admin")) {
             user.setRole("ROLE_ADMIN");
```
Comment (samuelprasad-yubi) [inline easy-auth-application/src/main/java/com/jmca/easyauthapplication/controller/AuthController.java]: @reviewzen suggestion

Comment (samuelprasad-yubi) [reply easy-auth-application/src/main/java/com/jmca/easyauthapplication/controller/AuthController.java]: @reviewzen suggestion

Comment (samuelprasad-yubi) [reply easy-auth-application/src/main/java/com/jmca/easyauthapplication/controller/AuthController.java]: @reviewzen suggestion

Comment (samuelprasad-yubi) [reply easy-auth-application/src/main/java/com/jmca/easyauthapplication/controller/AuthController.java]: @reviewzen suggestion

File: easy-auth-application/src/main/java/com/jmca/easyauthapplication/controller/AuthController.java [hunk 4]
```diff
@@ -78,9 +78,11 @@ public ResponseEntity<?> registerUser(@Valid @RequestBody SignupRequest signUpRe
                         encoder.encode(signUpRequest.getPassword()));
 
         if (signUpRequest.getRole() == null || signUpRequest.getRole().equalsIgnoreCase("user")) {
-            user.setRole("ROLE_USER");
-        } else if (signUpRequest.getRole().equalsIgnoreCase("admin")) {
-            user.setRole("ROLE_ADMIN");
+            user.setRole("ROLE_USE");
+} else if (signUpRequest.getRole().equalsIgnoreCase("admin")) {
```
Comment (samuelprasad-yubi) [inline easy-auth-application/src/main/java/com/jmca/easyauthapplication/controller/AuthController.java:82]: @reviewzen suggestion

File: easy-auth-application/src/main/java/com/jmca/easyauthapplication/controller/AuthController.java [hunk 5]
```diff
@@ -78,9 +78,11 @@ public ResponseEntity<?> registerUser(@Valid @RequestBody SignupRequest signUpRe
...
+    // Ensure the role is set to 'ROLE_ADMIN' in a case-insensitive manner
+    user.setRole(signUpRequest.getRole().toUpperCase() == "ADMIN" ? "ROLE_ADMIN" : "admin");
+}
         } else {
             throw new RuntimeException("Role Not Found!");
         }
```
Comment (samuelprasad-yubi) [inline easy-auth-application/src/main/java/com/jmca/easyauthapplication/controller/AuthController.java:88]: @reviewzen suggestion

Comment (samuelprasad-yubi) [reply easy-auth-application/src/main/java/com/jmca/easyauthapplication/controller/AuthController.java:88]: @reviewzen suggestion
